### PR TITLE
[MAINTENANCE] Typing render/view 

### PR DESCRIPTION
**Pull request by Kilo59**
Type checking for `render/view/view.py`

This change includes some runtime changes, notable the removal of the `NoOpTemplate` and the default `_template` class attribute for `DefaultJinjaView`

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -73,18 +73,24 @@ exclude = [
...
+    'core/usage_statistics/anonymizers/checkpoint_anonymizer\.py',       # 16 - This code will be removed in 1.0
+    'core/usage_statistics/anonymizers/data_docs_anonymizer\.py',        # 5 - This code will be removed in 1.0
+    'core/usage_statistics/anonymizers/datasource_anonymizer\.py',       # 9 - This code will be removed in 1.0
+    'core/usage_statistics/anonymizers/expectation_anonymizer\.py',      # 6 - This code will be removed in 1.0
+    'core/usage_statistics/anonymizers/validation_operator_anonymizer\.py',  # 5 - This code will be removed in 1.0
+    'render/renderer/v3/suite_edit_notebook_renderer\.py',               # 11 - This is legacy code and will not be typed.
+    'render/renderer/v3/suite_profile_notebook_renderer\.py',            # 4 - This is legacy code and will not be typed.
+    'render/renderer/suite_edit_notebook_renderer\.py',                  # 7 - This is legacy code and will not be typed.
+    'render/renderer/suite_scaffold_notebook_renderer\.py',              # 7 - This is legacy code and will not be typed.
+    'render/renderer/datasource_new_notebook_renderer\.py',              # 4 - This is legacy code and will not be typed.
```
Comment (Kilo59) [inline pyproject.toml:87]: @NathanFarmer do these look right? Am I missing anything?

Comment (billdirks) [reply pyproject.toml:87]: `great_expectations/render/renderer/notebook_renderer.py` is now typed but should it be deleted when we remove the cli?

**Review by billdirks — APPROVED** [review]
LGTM, ty!

File: pyproject.toml [hunk 2]
```diff
@@ -136,17 +142,10 @@ exclude = [
     'render/renderer/checkpoint_new_notebook_renderer\.py',                                                # 9
```
Comment (NathanFarmer) [inline pyproject.toml]: Missed this one

File: pyproject.toml (hunk 1 above)
Comment (NathanFarmer) [reply pyproject.toml:87]: Just missing: `render/renderer/checkpoint_new_notebook_renderer.py`

Comment (Kilo59) [reply pyproject.toml:87]: Good catch. Will update both.

Comment (NathanFarmer) [reply pyproject.toml:87]: I think `notebook_renderer.py` will also be removed with the CLI removal. It is the base class for the other notebook renderers.

**Review by cdkini — APPROVED** [review]

**Review by anthonyburdi — APPROVED** [review]
❤️

File: pyproject.toml (hunk 2 above)
Comment (Kilo59) [reply pyproject.toml]: Thanks!

**Pull request merged**